File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/virology/Country.java
package uk.nhs.nhsx.virology;

import uk.nhs.nhsx.core.ValueType;

public class Country extends ValueType<Country> {

    private Country(String value) {
        super(value);
    }

    public static Country of(String value) {
        return new Country(value);
    }
}

File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/virology/Country.kt
package uk.nhs.nhsx.virology

import dev.forkhandles.values.NonBlankStringValueFactory
import dev.forkhandles.values.StringValue

class Country private constructor(value: String) : StringValue(value) {
    companion object : NonBlankStringValueFactory<Country>(::Country) {
        val England = Country.of("England")
        val Wales = Country.of("Wales")
    }
}
